File: C/palindrome.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
/**
 * Palindromo
 * @author Vinicius Francisco da Silva
 * @version 1 08/2017
 */
bool igual(char* str);
int main(){
	char str[10000];
	for(scanf("%[^\n]\n",str); !igual(str); scanf("%[^\n]\n",str)){
		bool isPalindromo(char* str);
		bool resp = isPalindromo(str);
		if(resp == true){
			printf("SIM\n");
		}// End if
		else{
			printf("NAO\n");
		}// End else
		//setbuf(stdin, NULL);
	}// End for
	return 0;
}// End main()

/**
 * Verifica se é um palindromo ou não.
 * @return true caso seja um palindromo
 * false caso não seja
 */
bool isPalindromo(char* str){
	int i = 0;
	int j = strlen(str)-1;
	bool resp = false;
	while(i < j && j > i){
		if(str[i] == str[j]){
			resp = true;
		}// End if
		else{
			return false;
		}// End else
		i++;
		j--;
	}// End while
	return resp;
}// End ehPalindromo()

/**
 * Verifica se é a string que está como
 * parametro é igual a string "FIM".
 * @return true caso seja igual
 * false caso contrário
 */
bool igual(char str[]){
	bool resp = false;
	if(strlen(str) == 3){
		if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M'){
			resp = true;
		}// End if
		else{
			return false;
		}// End else
	}// End if
		else{
			return false;
		}// End else	
		return resp;
}// End igual()

File: Java/Palindrome.java
/**
 * Palindromo
 * @author Vinicius Francisco da Silva
 * @version 1 08/2017
 */
import java.util.Scanner;
public class Palindrome{
	public static void main(String[] args){
		String str;
		Scanner scanner = new Scanner(System.in);

		for(str = scanner.readLine(); !igual(str); str = scanner.readLine()){
			if(isPalindrome(str)){
				System.out.println("YES");
			}// End if
			else{
				System.out.println("NOT");
			}// End else
		}// End for
	}// End main()

	/*
     * Metodo publico iterativo para comparar a String
     * e retornar se ela é um palindromo ou não
     * @param String com a palavra que seja comparada
     * para descobrir se é ou não um palindromo
     * @return boolean identificando se é um palindromo
     * ou não
     */
	public static boolean isPalindrome(String str){
		int i = 0;
		int j = str.length()-1;
		boolean resp = false;
		while(i < j && j > i){
			if(str.charAt(i) == str.charAt(j)){
				resp = true;
			}// End if
			else{
				return false;
			}// End else
			i++;
			j--;
		}// End while
		return resp;
	}// End ehPalindromo()
	
	/*
     * Metodo publico para comparar a String
     * recebida como parametro com a String "FIM"
     * @param String com a palavra que seja comparada
     * @return boolean caso o parametro seja igual a
     * String "FIM"
     */
	public static boolean igual(String str){
		boolean resp = false;
		if(str.length() != 3){
			return resp;
		}// End if
		if(str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M'){
			resp = true;
		}// End if
		return resp;	
	}// End igual()
}// End class Palindromo
